bson/src/test/unit/org/bson/DocumentTest.java

```diff
@@ -108,6 +111,53 @@ public void shouldGetWithDefaultValue() {
         assertEquals("bar", z2);
     }
 
+    @Test
+    public void shouldGetListFromDocument() {
+        // given
+        Document d = new Document("x", 1)
+                .append("y", Arrays.asList("two", "three"))
+                .append("z", Arrays.asList(new Document("a", 1), new Document("b", 2)))
+                .append("w", new Document("a", Arrays.asList("One", "Two")))
+                .append("q", "notAList");
+
+        // when the value is a list
+        String three = d.getList("y", String.class).get(1);
+        int b = d.getList("z", Document.class).get(1).getInteger("b");
+        String two = d.get("w", Document.class).getList("a", String.class).get(1);
+
+        // extract the value from the list
+        assertEquals("three", three);
+        assertEquals(2, b);
+        assertEquals("Two", two);
+
+        // when the key is not found
+        List<String> nullValue = d.getList("yy", String.class);
+
+        // then it returns null
+        assertNull(nullValue);
+
+        // when the key is not found
+        List<String> defaultListValue = Arrays.asList("zero", "one");
+        List<String> listValue = d.getList("yy", String.class, defaultListValue);
+
+        // then it returns the default value
+        assertEquals(defaultListValue, listValue);
+    }
+
+    @Test
+    public void shouldThrowExceptionIfValueIsNotAList() {
```

Comment:
Can we also test retrieving a list of a different type? eg: `{a: [1, 2]}`  and `d.getList("a", String.class)`

Reply:
Done

Reply:
Included in the spock test